app/forms/incomeSources/cease/IncomeSourceEndDateForm.scala

```diff
@@ -37,7 +39,13 @@ class IncomeSourceEndDateForm @Inject()(val dateService: DateService) extends Cu
   val dateMustNotBeMissingDayAndMonthField = "dateForm.error.dayAndMonth.required"
   val dateMustNotBeMissingDayAndYearField = "dateForm.error.dayAndYear.required"
   val dateMustNotBeMissingMonthAndYearField = "dateForm.error.monthAndYear.required"
-  val dateMustNotBeInvalid = "error.invalid"
+  val dateInvalid = "error.invalid"
+
+  private def dateMustNotBeInvalid(incomeSourceType: IncomeSourceType) = {
```

Comment:
as discussed, we expect FS to be passed from the level of controller as it will be isolated on this level